File: Frontend/modules/dnd-module/ios/DndModule.swift
import ExpoModulesCore

public class DndModule: Module {
  public func definition() -> ModuleDefinition {
    Name("DndModule")

    Function("checkDndPermission") { () -> Bool in
      return false
    }

    Function("requestDndPermission") { () -> Void in
      // No-op on iOS
    }

    Function("setDndMode") { (enabled: Bool) -> Void in
      // No-op on iOS
    }

    Function("isDndEnabled") { () -> Bool in
      return false
    }
  }
}

File: Frontend/modules/dnd-module/src/DndModule.ts
import { NativeModule, requireNativeModule } from 'expo';

declare class DndModule extends NativeModule<{}> {
  checkDndPermission(): boolean;
  requestDndPermission(): void;
  setDndMode(enabled: boolean): void;
  isDndEnabled(): boolean;
  isNative: boolean;
}

let nativeModule: any = null;
try {
  nativeModule = requireNativeModule<DndModule>('DndModule');
  nativeModule.isNative = true;
} catch (e) {
  console.warn("DndModule native module not found (running in Expo Go/Simulator). Falling back to mock.");
  nativeModule = {
    checkDndPermission: () => false,
    requestDndPermission: () => {},
    setDndMode: (enabled: boolean) => {},
    isDndEnabled: () => false,
    isNative: false,
  };
}

export default nativeModule;
